01_Validation.R

```diff
@@ -0,0 +1,997 @@
+
+### . Change discharge reason from code to text
+
+table(datafile$discharge_to_code)
+
+datafile <- datafile %>% mutate(
+  discharge_to_code =
+    if_else(discharge_to_code %in% c("1", "01"), "Placement",
+      if_else(discharge_to_code %in% c("2", "02"), "Discharge Home with Home Care",
+        if_else(discharge_to_code %in% c("3", "03"), "Discharge Home",
+          if_else(discharge_to_code %in% c("4", "04"), "Death",
+            if_else(discharge_to_code %in% c("5", "05"), "Not Fit For Discharge", NA_character_))))))
+  
+table(datafile$discharge_to_code)
+count(datafile,discharge_to_code)
+# Check that variable length of postcode is 7 or less
+datafile <- datafile %>% mutate(
+  postcode_chars =
+    if_else(nchar(datafile$postcode) > 7, 1, 0))
+
+table(datafile$postcode_chars) # no postcodes with more than 7 characters
+
+# Fix formatting of postcode variable
+datafile <- datafile %>%
+
+  # First remove all spaces from postcode variable
+  mutate(
+    postcode = gsub("\\s", "", postcode),
+
+    # Then add space (or spaces) at appropriate juncture (depending on
+    # the number of characters) to get the postcode into 7-character
+    # format
+    postcode = case_when(
+      is.na(postcode) ~ NA_character_,
+      str_length(postcode) == 5 ~ sub("(.{2})", "\\1  ", postcode),
+      str_length(postcode) == 6 ~ sub("(.{3})", "\\1 ", postcode),
+      TRUE ~ postcode
+    )
+  )
+
+
+# datafile$FirstDoM <- month_start  # FirstDoM becomes month_start
+# datafile$LastDoM <- month_end # LastDoM becomes month_end
+
+# Create dob2 in yyyymmdd format
+datafile <- datafile %>%
+  mutate(dob2 = as.numeric(paste0(
+    str_sub(date_of_birth, 7, 10), str_sub(date_of_birth, 4, 5),
+    str_sub(date_of_birth, 1, 2)
+  )))
+
+# Create ready_medical_discharge_date as a string
+datafile <- datafile %>%
+  mutate(ready_medical_discharge_date = as.numeric(paste0(
+    str_sub(date_declared_medically_fit, 7, 10), str_sub(date_declared_medically_fit, 4, 5),
+    str_sub(date_declared_medically_fit, 1, 2)
+  )))
+
+
+# Compute Age at RMD Date ( Ready for Medical Discharge Date )
+
+datafile <- datafile %>%
+  mutate(age_at_rdd = trunc((ready_medical_discharge_date - dob2) / 10000))
+
+# Check there are no cases with a missing age_at_rdd
+df_missingage_at_rdd<-datafile %>% 
+  filter(is.na(age_at_rdd))
+# select if age_at_rdd>=18
+
+table(datafile$age_at_rdd)
+
+# amend dates to same formats
+datafile$date_declared_medically_fit <- format(as.Date(datafile$date_declared_medically_fit, "%d/%m/%Y"), "%Y/%m/%d")
+datafile$admission_date <- format(as.Date(datafile$admission_date, "%d/%m/%Y"), "%Y/%m/%d")
+datafile$discharge_date <- format(as.Date(datafile$discharge_date, "%d/%m/%Y"), "%Y/%m/%d")
+datafile$date_of_birth <- format(as.Date(datafile$date_of_birth, "%d/%m/%Y"), "%Y/%m/%d")
+month_start <- format(as.Date(month_start, "%d/%m/%Y"), "%Y/%m/%d")
+month_end <- format(as.Date(month_end, "%d/%m/%Y"), "%Y/%m/%d")
```

Comment:
Hi @petermcclurg,

I've had a look at the issue you had with dates formatted like 25-feb-2020 in the forth valley file. This is causing problems as in the above code chunk, dates are assumed to be in the format d/m/y.

I would suggest using the `lubridate` package to avoid this problem - it is a bit smarter than the `format` function as it'll recognise dates in a variety of formats as long as it's in the expected order; e.g. 

`dmy("25/02/2020")` will return `2020-02-25`, as will `dmy("25-feb-2020")` or `dmy("25/2/20")`, etc.

You could also make this section neater by using `mutate_at` to apply the same function to all variables with "date" in its name; e.g.

```suggestion
datafile %<>% mutate_at(vars(contains("date")), dmy)
```

Reply:
Hi @petermcclurg,
The line at the end of the above is to _replace_ the preceding lines - can you try removing the other lines that use `format` and rerun to see if the new line works then?